zarr/core.py

```diff
@@ -1489,7 +1489,7 @@ def _set_basic_selection_zd(self, selection, value, fields=None):
             chunk[selection] = value
 
         # encode and store
-        cdata = self._encode_chunk(chunk)
+        cdata = ensure_bytes(self._encode_chunk(chunk))
```

Comment:
N.B., ensuring bytes may introduce a memory copy, which may be unnecessary. Memory copies can noticeably affect performance. Ensuring we have something that supports the buffer interface would be better.